specs/interop.md

```diff
@@ -0,0 +1,188 @@
+based upgrade transactions are deterministically generated by the derivation pipeline in the following order:
+
+- L1 Attributes Transaction calling `setL1BlockValuesEcotone`
+- User deposits from L1
+- Network Upgrade Transactions
+  - L1Block deployment
+  - Update L1Block Proxy ERC-1967 Implementation Slot
+
+The execution payload MUST set `noTxPool` to `false` for this block.
+
+The exact definitions for these upgrade transactions are still to be defined.
+
+## Derivation Pipeline
+
+The derivation pipeline enforces invariants on safe blocks that include relaying transactions.
+
+- The relaying message MUST have a corresponding initiating message
+- The initiating message that corresponds to a relaying message MUST come from an allowed chain
+
+Blocks that contain transactions that relay cross domain messages to the remote chain where the
+initiating transaction does not exist MUST be considered invalid and MUST not be allowed by the
+derivation pipeline to be considered safe.
+
+There is no concept of replay protection within the protocol since there is no replay protection
+mechanism that fits well for all applications. Users MAY submit an arbitrary number of relaying
+messages per initiating message. This greatly simplifies the protocol and allows for
+applications to build application specific replay protection.
+
+### Depositing a Relaying Message
+
+Deposit transactions (force inclusion transactions) give censorship resistance to layer two networks.
+The derivation pipeline must gracefully handle the case in which a user uses a deposit transaction to
+relay a cross chain message. To not couple preconfirmation security to consensus, deposit transactions
+that relay cross chain messages MUST have an initiating message that is considered safe. This relaxes
+a strict synchrony assumption on the sequencer that it MUST have all unsafe blocks of remote chains
+as fast as possible to ensure that it is building correct blocks. It also prevents a class of attacks
+where the user can send a relaying message before the initiating message and trick the derivation pipeline
+into reorganizing the sequencer.
+
+### Safety
+
+The initiating messages for all relaying messages MUST be resolved as safe before an L2 block can transition from being unsafe to safe.
+Users MAY optimistically accept unsafe blocks without any verification of the relaying messages. They SHOULD optimistically verify
+the initiating messages exist in remote unsafe blocks to more quickly reorganize out invalid blocks.
+
+## Predeploys
+
+Two new system level predeploys are introduced for managing cross chain messaging.
+
+### CrossL2Outbox
+
+Address: `0x4200000000000000000000000000000000000023`
+
+The `CrossL2Outbox` is responsible for initiating a cross chain message on the local chain.
+For every cross chain message that is sent, it MUST emit an event that indicates the existence
+of the cross chain message.
+
+An initiating message 
```

Comment:
Missing a section?

Reply:
I cleaned this up, there may still be some dangling lines